src/design-system/components/menu/Menu.tsx

```diff
@@ -0,0 +1,181 @@
+  menuItems,
+  defaultOnItemClick,
+  maxHeight,
+  minWidth,
+  noItemsMessage = defaultNoItemsMessage,
+  density = 1,
+  allowToggleMultipleItems = false,
+  isItemsRemovalEnabled = false,
+  ...menuProps
+}: MenuProps) => {
+  const [items, setItems] = useState(menuItems)
+  const [toggledItemsTitles, setToggledItemsTitles] = useState<string[]>(
+    allowToggleMultipleItems
+      ? menuItems
+          .filter((item) => item.isInitiallyToggled)
+          .map((item) => item.title)
+      : []
+  )
+
+  const toggleAsSingleItem = (itemTitle: string) => {
+    setToggledItemsTitles((previousItems) =>
+      previousItems.includes(itemTitle) ? [] : [itemTitle]
+    )
+  }
+
+  const toggleAsOneOfMultipleItems = (itemTitle: string) => {
+    setToggledItemsTitles((previousItems) =>
+      previousItems.includes(itemTitle)
+        ? previousItems.filter((i) => i !== itemTitle)
+        : [...previousItems, itemTitle]
+    )
+  }
+
+  const handleMenuClose = useCallback(() => {
+    setToggledItemsTitles([])
+    setAnchorEl(null)
+  }, [setAnchorEl])
+
+  const handleItemClick = useCallback(
+    ({
+      title,
+      defaultOnItemClickArgs,
+      nestedMenuItems,
+      onClick: customOnClick
+    }: MenuItemProps) => {
+      allowToggleMultipleItems
+        ? toggleAsOneOfMultipleItems(title)
+        : toggleAsSingleItem(title)
+
+      if ((!customOnClick && !defaultOnItemClick) || nestedMenuItems) {
+        return
+      }
+
+      if (customOnClick) {
+        customOnClick()
+        handleMenuClose()
+      } else if (defaultOnItemClick) {
+        const args: OnItemClickArgs =
+          defaultOnItemClickArgs === undefined
+            ? { title }
+            : { title, ...defaultOnItemClickArgs }
+
+        defaultOnItemClick(args)
+      }
+
+      if (allowToggleMultipleItems) {
+        return
+      }
+
+      handleMenuClose()
+    },
+
+    [defaultOnItemClick, handleMenuClose, allowToggleMultipleItems]
+  )
+
+  const handleItemRemoval = (title: string) => {
+    setItems((previousItems) =>
+      previousItems.filter((item) => item.title !== title)
+    )
+  }
+
+  return (
+    <MuiMenu
+      anchorEl={anchorEl}
+      className={`s2s-menu s2s-menu--density-${density}`}
+      onClose={handleMenuClose}
+      open={Boolean(anchorEl)}
+      slotProps={{
+        paper: {
+          style: { maxHeight: maxHeight, minWidth: minWidth }
+        }
+      }}
+      {...menuProps}
+    >
+      {items.flatMap((item) => [
+        <MenuItem
+          {...item}
+          density={density}
+          isDropdown={Boolean(item.nestedMenuItems)}
+          isToggled={toggledItemsTitles.includes(item.title)}
+          key={item.title}
+          onClick={() => handleItemClick(item)}
+          onRemove={
+            isItemsRemovalEnabled
+              ? () => handleItemRemoval(item.title)
+              : undefined
+          }
+        />,
+        ...(item.nestedMenuItems && toggledItemsTitles.includes(item.title)
```

Comment:
I would swap those conditions to make them more understandable
```suggestion
        ...(toggledItemsTitles.includes(item.title) && item.nestedMenuItems
```